firmware/baseband/proc_replay.cpp

```diff
@@ -53,15 +53,20 @@ void ReplayProcessor::execute(const buffer_c8_t& buffer) {
     // 2048 samples * 2 bytes per sample = 4096 bytes
     // Since we're oversampling by 4M/500k = 8, we only need 2048/8 = 256 samples from the file and duplicate them 8 times each
     // So 256 * 4 bytes per sample (C16) = 1024 bytes from the file
+
+    // Wrap the IQ data array in a buffer with the correct sample_rate.
+    buffer_c16_t iq_buffer{iq.data(), iq.size(), baseband_fs / toUType(oversample_rate)};
+
     const size_t bytes_to_read = sizeof(*buffer.p) * 2 * (buffer.count / 8);  // *2 (C16), /8 (oversampling) should be == 1024
     size_t bytes_read_this_iteration = stream->read(iq_buffer.p, bytes_to_read);
     size_t oversamples_this_iteration = bytes_read_this_iteration * 8 / (sizeof(*buffer.p) * 2);
 
     bytes_read += bytes_read_this_iteration;
 
+    // TODO: fix for oversample_rate
     // Fill and "stretch"
     for (size_t i = 0; i < oversamples_this_iteration; i++) {
-        if (i & 7) {
+        if (i > 0) {
```

Comment:
Looks like this will fill the entire transmit buffer with the first sample.  (Perhaps "oversamples_this_iteration" is not a good variable name, because this loop is actually going through the whole buffer.)